src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,85 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FC, memo } from 'react';
+import { FilterBy } from '../../types/FilterBy';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  query: string;
+  onChangeQuery: (event: string) => void;
+  onChangeApplyQuery : (event: string) => void;
+  onChangeOption: (event: FilterBy) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: FC<Props> = memo((
+  {
+    query,
+    onChangeOption,
+    onChangeQuery,
+    onChangeApplyQuery,
+  },
+) => {
+  const handleClearQuery = () => onChangeQuery('');
```

Comment:
It's a good practice to add a type to the 'handleClearQuery' function: 'const handleClearQuery: () => void = () => ...'